build/upload-functionsZipToStorageAccount.ps1

```diff
@@ -0,0 +1,198 @@
+param (
+    [Parameter(Mandatory=$true)]
+    [ValidateNotNullOrEmpty()]
+    [System.String]
+    $StorageAccountName,
+
+    [Parameter(Mandatory=$true)]
+    [ValidateNotNullOrEmpty()]
+    [System.String]
+    $StorageAccountKey,
+
+    [Parameter(Mandatory=$true)]
+    [ValidateNotNullOrEmpty()]
+    [System.String]
+    $SourcePath
+)
+
+function WriteLog
+{
+    param (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Message,
+
+        [Switch]
+        $Throw
+    )
+
+    $Message = (Get-Date -Format G)  + " -- $Message"
+
+    if ($Throw)
+    {
+        throw $Message
+    }
+
+    Write-Host $Message
+}
+
+function GetContentType
+{
+    param
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $FilePath
+    )
+
+    $fileExtension =  [System.IO.Path]::GetExtension($FilePath)
+
+    switch ($fileExtension)
+    {
+        ".txt" { "text/plain" }
+        ".json" { "application/json" }
+        default { "application/octet-stream"}
+    }
+}
+
+function GetFullPath
+{
+    param
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $FilePath
+    )
+
+    $FilePath = (Get-Item $FilePath -ErrorAction SilentlyContinue).FullName
+
+    if (-not $FilePath)
+    {
+        WriteLog -Message "Required file '$FilePath' does not exist." -Throw
+    }
+
+    return $FilePath
+}
+
+function GetFunctionsBuildVersion
+{
+    $filePath = GetFullPath -FilePath "$SourcePath/Functions.*.zip"
+    $fileName = Split-Path -Path $filePath -Leaf
+
+    $pattern = 'Functions.(.+).zip'    
+    $version = $fileName -replace $pattern,'$1'
+
+    if (-not $version)
+    {
+        WriteLog -Message "Failed to parse version string from '$fileName'." -Throw
+    }
+
+    return $version
+}
+
+WriteLog -Message "Script started."
+
+if (-not (Test-Path $SourcePath))
+{
+    throw "SourcePath '$SourcePath' does not exist."
+}
+
+$CONTAINER_NAME = "functionsbuilds"
+$FUNC_RUNTIME_VERSION = '3'
```

Comment:
this should input param ? as we need to generate and upload nightly builds for V4 as well?